### PR TITLE
feat: run status checks on pr

### PR DESCRIPTION
**Pull request by dkelosky**
Signed-off-by: Dan Kelosky <dkelosky@gmail.com>

## Proposed changes

Renames the build for theia pipeline to `theia-build` so that the status check is more descriptive than just `build`:

image

## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] `npm run vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Comment by dkelosky** [comment]
@zFernand0 - youll have to update the required check to `theia-build` 

@tjohnsonBCM & @awharn - potentially we'll want `on: [push, pull_request]` added to imperative and zowe-cli repos.  assuming this works by allowing status checks from forks.

**Comment by zFernand0** [comment]
> youll have to update the required check to `theia-build` 

Done!

**Comment by awharn** [comment]
> @zFernand0 - youll have to update the required check to `theia-build` 
> 
> @tjohnsonBCM & @awharn - potentially we'll want `on: [push, pull_request]` added to imperative and zowe-cli repos.  assuming this works by allowing status checks from forks.

We want to do this. One consideration is that if you do work on a branch in the repo and create a PR, each build runs twice: once for the branch and once for the PR. This will exhaust all the runners for Imperative and the CLI and not run any more actions until they free up (approx 20-30 minutes). If someone pushes commits frequently, it will take a considerable length of time until the runners free up again.

**Comment by dkelosky** [comment]
i think it only seems to run once.  From testing, it looks like the checks run on a push and are not re-run on when the PR is opened.  (the checks are immediately marked as passed)

**Comment by awharn** [comment]
> I think it only seems to run once. From testing, it looks like the checks run on a push and are not re-run on when the PR is opened. (the checks are immediately marked as passed)

Odd. I am not seeing that behavior on [my PR](https://github.com/zowe/imperative/pull/430).

**Comment by dkelosky** [comment]
i did it at a slower sequence.
- push, wait for status checks to finish, then open the PR.

**Comment by awharn** [comment]
> i did it at a slower sequence.
> 
> * push, wait for status checks to finish, then open the PR.

Oh, that makes more sense. I believe this behavior will still occur if a PR is open and a person pushes another change to the branch, like in the case of a change in response to a peer review. There might be a way around that, though...

**Comment by dkelosky** [comment]
hmm... i think you're right

**Comment by zFernand0** [comment]
> > youll have to update the required check to `theia-build`
> 
> Done!

FYI, All PRs will have an always pending status-check until this PR gets merged.

**Comment by dkelosky** [comment]
> FYI, All PRs will have an always pending status-check until this PR gets merged.

maybe we can get some reviewers then 😄 ?

File: .github/workflows/theia-zowe-explorer-ci.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Workflow name for GitHub Actions
 name: Theia Zowe Explorer CI
 
-on: [push]
+on: [push, pull_request]
 
 jobs:
 
-  build:
+  thiea-build:
```
Comment (zFernand0) [inline .github/workflows/theia-zowe-explorer-ci.yml:8]: should it be theia?

Comment (dkelosky) [reply .github/workflows/theia-zowe-explorer-ci.yml]: embarrassing ... 😢

**Pull request merged**